crates/pathfinder/src/state/sync/l1.rs

```diff
@@ -1,680 +1,49 @@
-use anyhow::Context;
-use ethers::types::H160;
-use futures::Future;
-use pathfinder_common::{BlockNumber, Chain, EthereumBlockHash, EthereumBlockNumber};
-use pathfinder_ethereum::{
-    log::StateUpdateLog,
-    provider::EthereumTransport,
-    state_update::{FetchError, StateRootFetcher},
-};
-use pathfinder_retry::Retry;
-use std::{num::NonZeroU64, sync::Arc, time::Duration};
-use tokio::sync::{mpsc, oneshot, RwLock};
+use std::num::NonZeroU64;
 
-/// Events and queries emitted by L1 sync process.
-#[derive(Debug)]
-pub enum Event {
-    /// New L1 [update logs](StateUpdateLog) found.
-    Update(Vec<StateUpdateLog>),
-    /// An L1 reorg was detected, contains the reorg-tail which
-    /// indicates the oldest block which is now invalid
-    /// i.e. reorg-tail + 1 should be the new head.
-    Reorg(BlockNumber),
-    /// Query for the [update log](StateUpdateLog) of the given block.
-    ///
-    /// The receiver should return the [update log](StateUpdateLog) using the
-    /// [oneshot::channel].
-    QueryUpdate(BlockNumber, oneshot::Sender<Option<StateUpdateLog>>),
-}
+use pathfinder_common::Chain;
+use pathfinder_ethereum::{EthereumApi, EthereumStateUpdate};
+use pathfinder_retry::Retry;
+use primitive_types::H160;
+use tokio::sync::mpsc;
 
-/// Syncs L1 state update logs. Emits [sync events](Event) which should be handled
-/// to update storage and respond to queries.
+/// Syncs L1 state update logs. Emits [Ethereum state update](EthereumStateUpdate)
+/// which should be handled to update storage and respond to queries.
 pub async fn sync<T>(
-    tx_event: mpsc::Sender<Event>,
-    transport: T,
+    tx_event: mpsc::Sender<EthereumStateUpdate>,
+    ethereum: T,
     chain: Chain,
     core_address: H160,
-    head: Option<StateUpdateLog>,
 ) -> anyhow::Result<()>
 where
-    T: EthereumTransport + Send + Sync + Clone,
+    T: EthereumApi + Send + Sync + Clone,
```

Comment:
Bounds are more restrictive than `sync_impl`

Reply:
I didn't change the bounds, only the trait name.

Reply:
Relaxed the restricted version to match `sync`.